amethyst_core/src/transform/systems.rs

```diff
@@ -39,11 +48,15 @@ impl<'a> System<'a> for TransformSystem {
     type SystemData = (
         Entities<'a>,
         ReadExpect<'a, ParentHierarchy>,
-        ReadStorage<'a, Transform>,
+        ReadStorage<'a, IsometricTransform>,
+        WriteStorage<'a, Transform>,
```

Comment:
This approach was suggested by Xaeroxe, because it is not possible to set up dependencies in an optional system.

Reply:
Yes it is. You have to do it in the bundle.

Reply:
I want to reuse the scratch to automatically populate Transform tho.

Reply:
More parallelism and clarity is always the reason I separate systems.

Reply:
Yeah, and I don't think it beats reusing the scratch here. Most notably because this grabs a WriteStorage on Transform. Besides, it's pretty clear what it does here.